### ISSUE TITLE
Add config to make link interception opt-in

### ISSUE THREAD

**Issue by brillout** [brillout/vite-plugin-ssr#918]
### Description

By default, when using Client Routing, vite-plugin-ssr automatically intercepts links. You can opt-out with `rel="external"` (e.g. `<a href="/admin" rel="external">`).

Instead of opt-out, there could be a new config `disableAutomaticLinkInterception`, together with a new attribute `data-vike-link` to opt into Client Routing (e.g. `<a href="/product/42" data-vike-link>`).

The config can be set in a non-global fashion, i.e. each page can have a different `disableAutomaticLinkInterception` setting.

```js
// /dashboard/+config.js

// The links displayed at URL `/dashboard` *aren't* intercepted 
export default {
  disableAutomaticLinkInterception: true
}
```

```js
// /product/@id/+config.js

// The links displayed at URL `/product/42` *are* intercepted 
export default {
  disableAutomaticLinkInterception: false
}
```

**Comment by jahredhope** [comment]
I think this feature would be really valuable.

I raised my reasoning previously in https://github.com/brillout/vite-plugin-ssr/discussions/728

**Comment by brillout** [comment]
ETA this week.

**Comment by brillout** [comment]
Implemented. Set `window._disableAutomaticLinkInterception = true` and use `<a data-vike-link>`.

> :warning: `_disableAutomaticLinkInterception` will be removed in the next major release. Because, in the next major release, the link interceptor will be able to check whether the URL matches the route of a page, making `_disableAutomaticLinkInterception` obsolete. (The change requires a breaking change and that's why it can't be released already as `v0.4.x`.)

**Comment by brillout** [comment]
Pre-released in `0.4.131-commit-35ca471f7`.
